test_runner/regress/test_download_extensions.py

```diff
@@ -0,0 +1,112 @@
+
+                with open(full_path, "rb") as f:
+                    log.info(f"UPLOAD {full_path} to ext/{full_path}")
+                    env.remote_storage_client.upload_fileobj(
+                        f,
+                        env.ext_remote_storage.bucket_name,
+                        f"ext/{full_path}",
+                    )
+        os.chdir("../../../..")
+    # Start a compute node and check that it can download the extensions
+    # and use them to CREATE EXTENSION and LOAD
+    endpoint = env.endpoints.create_start(
+        "test_remote_extensions",
+        tenant_id=tenant_id,
+        remote_ext_config=env.ext_remote_storage.to_string(),
+        # config_lines=["log_min_messages=debug3"],
+    )
+    with closing(endpoint.connect()) as conn:
+        with conn.cursor() as cur:
+            # Check that appropriate control files were downloaded
+            cur.execute("SELECT * FROM pg_available_extensions")
+            all_extensions = [x[0] for x in cur.fetchall()]
+            log.info(all_extensions)
+            assert "anon" in all_extensions
+            assert "embedding" in all_extensions
+            # TODO: check that we cant't download custom extensions for other tenant ids
+
+            # check that we can download public extension
+            cur.execute("CREATE EXTENSION embedding")
+            cur.execute("SELECT extname FROM pg_extension")
+            assert "embedding" in [x[0] for x in cur.fetchall()]
+
+            # check that we can download private extension
+            try:
+                cur.execute("CREATE EXTENSION anon")
+            except Exception as err:
+                log.info("error creating anon extension")
+                assert "pgcrypto" in str(err), "unexpected error creating anon extension"
+
+            # TODO: try to load libraries as well
+
+    # Cleaning up downloaded files is important for local tests
+    # or else one test could reuse the files from another test or another test run
+    cleanup_files = [
+        "lib/postgresql/anon.so",
+        "lib/postgresql/embedding.so",
+        "share/postgresql/extension/anon.control",
+        "share/postgresql/extension/embedding--0.1.0.sql",
+        "share/postgresql/extension/embedding.control",
+    ]
+    cleanup_files = [f"pg_install/v{pg_version}/" + x for x in cleanup_files]
+    cleanup_folders = [
+        "extensions",
+        f"pg_install/v{pg_version}/share/postgresql/extension/anon",
+        f"pg_install/v{pg_version}/extensions",
+    ]
+    for file in cleanup_files:
+        try:
+            os.remove(file)
+            log.info(f"removed file {file}")
+        except Exception as err:
+            log.info(f"error removing file {file}: {err}")
+    for folder in cleanup_folders:
+        try:
+            shutil.rmtree(folder)
+            log.info(f"removed folder {folder}")
+        except Exception as err:
+            log.info(f"error removing file {file}: {err}")
```

Comment:
I get an error, when run the test:
```
2023-07-21 16:10:18.118 INFO [test_download_extensions.py:104] removed file pg_install/v15/share/postgresql/extension/embedding.control
2023-07-21 16:10:18.118 INFO [test_download_extensions.py:112] error removing file pg_install/v15/share/postgresql/extension/embedding.control: [Errno 2] No such file or directory: 'extensions'
2023-07-21 16:10:18.119 INFO [test_download_extensions.py:110] removed folder pg_install/v15/share/postgresql/extension/anon
2023-07-21 16:10:18.119 INFO [test_download_extensions.py:112] error removing file pg_install/v15/share/postgresql/extension/embedding.control: [Errno 2] No such file or directory: 'pg_install/v15/extensions'
```

Reply:
the error indicates that the files failed to download.
real s3 tests are expected to fail now, until the bucket infra agrees with the rust logic, was it a real s3 test?

it looks like these are paths should be right. can you see if the files were downloaded to some other incorrect location?
you can check `compute.log` it should say where it creates files.

Reply:
(the tests work for me, and the mock_s3 tests pass on CI last I checked)